src/vma/dev/net_device_table_mgr.cpp

```diff
@@ -323,10 +333,119 @@ void net_device_table_mgr::verify_ipoib_mode(struct ifaddrs* ifa)
+
+	qp_init_attr.cap.max_send_wr = safe_mce_sys().tx_num_wr;
+	qp_init_attr.cap.max_recv_wr = safe_mce_sys().rx_num_wr;
+	qp_init_attr.cap.max_inline_data = safe_mce_sys().tx_max_inline;
+	qp_init_attr.cap.max_send_sge = MCE_DEFAULT_TX_NUM_SGE;
+	qp_init_attr.cap.max_recv_sge = MCE_DEFAULT_RX_NUM_SGE;
+	qp_init_attr.sq_sig_all = 0;
+	qp_init_attr.qp_type = qp_type;
+
+	//find ib_cxt
+	char base_ifname[IFNAMSIZ];
+	get_base_interface_name((const char*)(ifname), base_ifname, sizeof(base_ifname));
+	struct ibv_context** pp_ibv_context_list = rdma_get_devices(&num_devices);
+	char resource_path[256];
+	sprintf(resource_path, VERBS_DEVICE_RESOURCE_PARAM_FILE, base_ifname);
+	char sys_res[1024] = {0};
+	priv_safe_read_file(resource_path, sys_res, 1024);
+	for (int j=0; j<num_devices; j++) {
+		char ib_res[1024] = {0};
+		char ib_path[256] = {0};
+		sprintf(ib_path, "%s/device/resource", pp_ibv_context_list[j]->device->ibdev_path);
+		priv_safe_read_file(ib_path, ib_res, 1024);
+		if (strcmp(sys_res, ib_res) == 0) {
+			//create qp resources
+			ib_ctx_handler* p_ib_ctx = g_p_ib_ctx_handler_collection->get_ib_ctx(pp_ibv_context_list[j]);
+			struct ibv_comp_channel *channel = ibv_create_comp_channel(p_ib_ctx->get_ibv_context());
+			if (!channel) {
+				ndtm_logdbg("channel creation failed for interface %s (errno=%d %m)", ifname, errno);
+				break;
+			}
+			struct ibv_cq* cq = vma_ibv_create_cq(p_ib_ctx->get_ibv_context(), safe_mce_sys().tx_num_wr, (void*)this, channel, 0, &attr);
+			if (!cq) {
+				ndtm_logdbg("cq creation failed for interface %s (errno=%d %m)", ifname, errno);
+				break;
+			}
+			qp_init_attr.recv_cq = cq;
+			qp_init_attr.send_cq = cq;
+			struct ibv_qp* qp = ibv_create_qp(p_ib_ctx->get_ibv_pd(), &qp_init_attr);
+			if (qp) {
+				success = true;
+				IF_VERBS_FAILURE(ibv_destroy_qp(qp)) {
+					ndtm_logdbg("qp destroy failed on interface %s (errno=%d %m)", ifname, errno);
+				} ENDIF_VERBS_FAILURE;
+				IF_VERBS_FAILURE(ibv_destroy_cq(cq)) {
+					ndtm_logdbg("cq destroy failed on interface %s (errno=%d %m)", ifname, errno);
+				} ENDIF_VERBS_FAILURE;
+				IF_VERBS_FAILURE(ibv_destroy_comp_channel(channel)) {
+					ndtm_logdbg("channel destroy failed on interface %s (errno=%d %m)", ifname, errno);
+				} ENDIF_VERBS_FAILURE;
+			}
+			break;
+		}
+	}
+	rdma_free_devices(pp_ibv_context_list);
+	if (!success) {
+		int priv = validate_raw_qp_privliges();
+		if (priv == -1) {
+			//file doesn't exist, check cap_net_raw
+			if (!check_user_capabilities()) {
+				vlog_printf(VLOG_WARNING,"******************************************************************************\n");
+				vlog_printf(VLOG_WARNING,"* Usage is restricted to root or CAP_NET_RAW privileges.                     *\n");
+				vlog_printf(VLOG_WARNING,"* Interface %s will not be offloaded.                                         \n", ifname);
+				vlog_printf(VLOG_WARNING,"******************************************************************************\n");
```

Comment:
@dinal I agree with you comment. So please extend the PR to include update of README to better explain both the upstream and MLNX_OFED configuration options